contentcuration/contentcuration/static/js/edit_channel/channel_set/util.js

```diff
@@ -0,0 +1,56 @@
+var Models = require('../models');
+const State = require("edit_channel/state");
+
+exports.loadChannelSetChannels = function(token) {
+  return new Promise(function(resolve, reject) {
+    $.ajax({
+      method:"GET",
+      url: window.Urls.get_channels_by_token(token),
+      success: resolve,
+      error: reject
+    });
+  })
+}
+
+exports.loadChannelList = function(url) {
+  return new Promise(function(resolve, reject) {
+    $.ajax({
+      method:"GET",
+      url: url,
+      success: resolve,
+      error: reject,
+      data: {
+        "published": true
+      }
+    });
+  })
+}
+
+exports.saveChannelsToSet = function(channelSetData, channels) {
+  return new Promise(function(resolve, reject) {
+    let channelSetModel = new Models.ChannelSetModel();
+    delete channelSetData.secret_token; // Don't save this
+
+    if(!channelSetData.editors || channelSetData.editors.length === 0) {
+      channelSetData.editors = [State.current_user.id];
+    }
+
+    // Need to save first to make sure token is available
+    channelSetModel.save(channelSetData, {
+      patch: true,
+      error: reject,
+      success: function(channelSet) {
+        $.ajax({
+          method:"POST",
+          url: window.Urls.save_token_to_channels(channelSet.get('secret_token').token),
+          success: function() {
+            channelSet.fetch({success: resolve});
```

Comment:
We seem to be calling three Ajax calls here, and I feel like there is no additional data being returned to the frontend after the first call. Could we modify the create method of the ChannelSet serializer to do this work for us, rather than create three requests?

Reply:
Is there a way to pass in extra arguments from backbone to the serializer? When I try adding a `channels` field on the channel set, it doesn't persist to the `validated_data` argument on the serializer

Reply:
The easiest way to do that is just to add additional attributes to the model before saving. If you don't want those arguments on the model itself, you can invoke `Backbone.sync` directly (see the default save method here: http://backbonejs.org/docs/backbone.html) and add extra arguments to the payload.

Reply:
Updated! It turns out the serializer was the one that was removing the data, but the channels were in the `initial_data` attribute